sound/soc/sof/ipc.c

```diff
@@ -184,6 +194,8 @@ struct snd_sof_ipc *snd_sof_ipc_init(struct snd_sof_dev *sdev)
 		return NULL;
 	}
 
+	ipc->ops = ops;
```

Comment:
if neither CONFIG_SND_SOC_SOF_IPC3 nor CONFIG_SND_SOC_SOF_INTEL_IPC4 are chosen, can ops be NULL

Reply:
it will hit the `default:` case. This is why I have kept the `case .... :` within the ifdef